### DISCUSSION TITLE
OfflineRigidbody - Add a Getter for the `RigidbodyPauser`

### DISCUSSION THREAD

**Discussion by AustinEast** [FirstGearGames/FishNet#756] [Feature Request]
I've recently found the need to access the `RigidbodyPauser` in my Offline Rigidbodies to work with my pooling system - my projectiles have the possibility of getting spawned from the pool with their rigidbodies paused, but i need to be able to set their velocity in that moment, so heres what works for me:

```cs

// Pooled Projectile Spawning logic...

...

if (projectileObject.TryGetComponent<OfflineRigidbody>(out var offlineRigidbody))
{
    var rigidbodyPauser = offlineRigidbody.GetRigidbodyPauser();
    rigidbodyPauser.Unpause();
}

if (projectileObject.TryGetComponent<Rigidbody>(out var projectileRigidbody) && !projectileRigidbody.isKinematic)
{
    projectileRigidbody.velocity = direction * gunSettings.projectileSettings.speed;
}

if (PredictionManager.IsReconciling && offlineRigidbody != null)
{
    var rigidbodyPauser = offlineRigidbody.GetRigidbodyPauser();
    rigidbodyPauser.Pause();
            
```

My request is to add a getter like this to make this possible:

```cs
public partial class OfflineRigidbody : MonoBehaviour
{
   ...

    public RigidbodyPauser GetRigidbodyPauser() => _rigidbodyPauser;
}
```

**Comment by FirstGearGames** [comment]
Doesn't this already exists?

**Reply by AustinEast** [reply]
I dont think it does, unless I'm missing something. The `_rigidbodyPauser` field is private, at least.

**Comment by FirstGearGames** [comment]
This was added some releases ago after your request. Closing :)

**Comment by AustinEast** [comment]
I don't see where it's been added in `OfflineRigidbody`, am i missing a partial class somewhere?

**Comment by FirstGearGames** [comment]
Oh offline rigidbody, sorry. It has not been added. I will expose it but I
can probably do you one better though next update or the one after that.
